### PR TITLE
Semaphore 2.0 config file.

### PR DESCRIPTION
**Pull request by Lord-Kamina**
### What does this PR do?

Add the initial sempahore-2.0 config file.

### More

- [ ] Added/updated documentation

### CODE REVIEW

**Comment by Baklap4** [comment]
I do like the way it is defined though, very much gitlab alike

**Comment by Lord-Kamina** [comment]
Yeah I was just reading the reference (took me a bit to find it). Looks nice.

**Comment by Baklap4** [comment]
I'd say the steps will be:
Fetch dependencies (parallel for each system (linux, unix, win32 (cross/native)))
Build (parallel for each system (linux, unix, win32 (cross/native)))
Deploy all versions as release to github (when master is targeted) (not sure if this is nicely integrated in github tho)

**Comment by Lord-Kamina** [comment]
That'd be great in the long run but I don't think it's possible for now at least. Only Ubuntu 18.04 is available for now.

**Comment by Baklap4** [comment]
with ubuntu 18 its possible to at least do 4/6 iems by crosscompiling. however we need to get rid of cpprestsdk first then :o

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -12,9 +12,9 @@ blocks:
         - name: Get dependencies.
           commands:
              - checkout
-             - sudo apt-get -y update
+             - sudo apt -y update
              - sudo apt-get -y install cmake gettext help2man clang gcc-7 g++-7 libepoxy-dev libsdl2-dev libcairo2-dev libpango1.0-dev librsvg2-dev libboost-all-dev libavcodec-dev libavformat-dev libswscale-dev libswresample-dev libpng-dev libjpeg-dev libxml++2.6-dev portaudio19-dev libopencv-dev libportmidi-dev libjsoncpp-dev libqrencode-dev libicu-dev libcpprest-dev libglm-dev
```
Comment (Baklap4) [inline .semaphore/semaphore.yml]: Also use `sudo apt install -y package-1 package-2 package-3`

Comment (Lord-Kamina) [reply .semaphore/semaphore.yml]: Will change it later. Anyway this was made to try and get around the insufficient space issue which is completely a problem on semaphore's side.

Comment (Baklap4) [reply .semaphore/semaphore.yml]: I thought so hahah. Hmm is there an issue or such which we can follow?

Comment (Lord-Kamina) [reply .semaphore/semaphore.yml]: I've tracked the issue down to libsdl2, which requires udev and that in turn wants a new kernel, which causes the error with initramfs-tools. Still, that would be solved by making the /boot partition larger. I've made a thread on the semaphore Slack so if you've got Slack you can follow it there.

**Comment by nieknooijens** [comment]
did @Lord-Kamina force-push or something? since the comments from @Baklap4 are outdated yet there's only 1 commit.

**Comment by Lord-Kamina** [comment]
@nieknooijens yes I rebased everything into a single commit and force-pushed.

**Comment by Baklap4** [comment]
Approving and linked #389 to this to upgrade our ci to be more useful for us :)

**Pull request merged**